contracts/fio.address/fio.address.cpp

```diff
@@ -669,6 +727,134 @@ namespace fioio {
+            fio_403_assert(fioname_iter->owner_account == actor.value,
+                           ErrorSignature); // check if actor owns FIO Address
+
+
+
+
+            //FEE PROCESSING
+            uint64_t fee_amount = 0;
+
+            //begin fees, bundle eligible fee logic
+            const uint128_t endpoint_hash = string_to_uint128_hash(UPDATE_ENCRYPT_KEY_ENDPOINT);
+
+            auto fees_by_endpoint = fiofees.get_index<"byendpoint"_n>();
+            auto fee_iter = fees_by_endpoint.find(endpoint_hash);
+
+            //if the fee isnt found for the endpoint, then 400 error.
+            fio_400_assert(fee_iter != fees_by_endpoint.end(), "endpoint_name", REMOVE_PUB_ADDRESS_ENDPOINT,
+                           "FIO fee not found for endpoint", ErrorNoEndpoint);
+
+            const int64_t reg_amount = fee_iter->suf_amount;
+            const uint64_t fee_type = fee_iter->type;
+
+            fio_400_assert(fee_type == 1, "fee_type", to_string(fee_type),
+                           "update_encrypt_key unexpected fee type for endpoint update_encrypt_key, expected 1",
+                           ErrorNoEndpoint);
+
+            const uint64_t bundleeligiblecountdown = fioname_iter->bundleeligiblecountdown;
+
+            if (bundleeligiblecountdown > 0) {
+                namesbyname.modify(fioname_iter, _self, [&](struct fioname &a) {
+                    a.bundleeligiblecountdown = (bundleeligiblecountdown - 1);
+                });
+            } else {
+                fee_amount = fee_iter->suf_amount;
+                fio_400_assert(max_fee >= (int64_t) fee_amount, "max_fee", to_string(max_fee),
+                               "Fee exceeds supplied maximum.",
+                               ErrorMaxFeeExceeded);
+
+                //NOTE -- question here, should we always record the transfer for the fees, even when its zero,
+                //or should we do as this code does and not do a transaction when the fees are 0.
+                fio_fees(actor, asset(reg_amount, FIOSYMBOL), REMOVE_PUB_ADDRESS_ENDPOINT);
+                process_rewards(tpid, reg_amount, get_self(), actor);
+
+                if (reg_amount > 0) {
+                    INLINE_ACTION_SENDER(eosiosystem::system_contract, updatepower)
+                            ("eosio"_n, {{_self, "active"_n}},
+                             {actor, true}
+                            );
+                }
+            }
+
+            if (UPDENCRYPTKEYRAM > 0) {
+                action(
+                        permission_level{SYSTEMACCOUNT, "active"_n},
+                        "eosio"_n,
+                        "incram"_n,
+                        std::make_tuple(actor, UPDENCRYPTKEYRAM)
+                ).send();
+            }
+
+            updhandleinf(encrypt_public_key, FIO_REQUEST_CONTENT_ENCRYPTION_PUB_KEY_DATA_DESC,fioname_iter->id,actor);
+
+            fio_400_assert(fee_iter != fees_by_endpoint.end(), "endpoint_name", REMOVE_PUB_ADDRESS_ENDPOINT,
```

Comment:
Why is this assertion duplicated here? It is also at line 797.

Should be UPDATE_ENCRYPT_KEY_ENDPOINT

Reply:
yes....vestigial code, is removed